src/testing/testpachd/main.go

```diff
@@ -31,37 +31,37 @@ var (
 )
 
 func main() {
+	var err error
 	flag.Parse()
 
-	// Init logging.
-	done := log.InitBatchLogger(*logfile)
+	defer func(done func(error)) {
+		done(err)
+	}(log.InitBatchLogger(*logfile))
 	if *verbose {
 		log.SetLevel(log.DebugLevel)
 	}
 
-	// Handle cleaning up on exit.
-	ctx, cancel := pctx.Interactive()
-	runCtx := ctx
-	var exitErr error
 	var clean cleanup.Cleaner
 	defer func() {
-		ctx = pctx.Background("cleanup")
-		if err := clean.Cleanup(ctx); err != nil {
+		ctx := pctx.Background("cleanup")
+		if err = clean.Cleanup(ctx); err != nil {
```

Comment:
I think an error when cleaning up is probably not worth deleting the batch logs?